projectile.el

```diff
@@ -678,6 +679,13 @@ which we're looking."
              (setq file nil))))
     (and root (expand-file-name (file-name-as-directory root)))))
 
+(defvar-local projectile-custom-root nil
+  "Defines a custom Projectile project root.
+   This is intended to be used as a file local variable.")
+
+(defun projectile-root-file-local (dir)
```

Comment:
This ended up pretty severely regressing performance if `projectile-project-root` is nil because it does a `file-truename` on `dir` every time.